File: scala-akka-http-server/src/main/scala/org/openapitools/server/model/Bot.scala
package org.openapitools.server.model


/**
 * @param userId  for example: ''null''
 * @param fullName  for example: ''null''
 * @param apiKey  for example: ''null''
 * @param defaultSendingStream  for example: ''null''
 * @param defaultEventsRegisterStream  for example: ''null''
 * @param defaultAllPublicStreams  for example: ''null''
 * @param avatarUrl  for example: ''null''
 * @param ownerId  for example: ''null''
 * @param services  for example: ''null''
 * @param email The email of the bot.  for example: ''null''
 * @param botType An integer describing the type of bot: * `1` for a `Generic` bot. * `2` for an `Incoming webhook` bot. * `3` for an `Outgoing webhook` bot. * `4` for an `Embedded` bot.  for example: ''null''
 * @param isActive A boolean describing whether the user account has been deactivated.  for example: ''null''
*/
final case class Bot (
  userId: Option[AnyType],
  fullName: Option[AnyType],
  apiKey: Option[AnyType],
  defaultSendingStream: Option[AnyType],
  defaultEventsRegisterStream: Option[AnyType],
  defaultAllPublicStreams: Option[AnyType],
  avatarUrl: Option[AnyType],
  ownerId: Option[AnyType],
  services: Option[AnyType],
  email: Option[String],
  botType: Option[Int],
  isActive: Option[Boolean]
)

File: fsharp-functions/OpenAPI/src/model/Bot.fs
namespace OpenAPI.Model

open System
open System.Collections.Generic
open Newtonsoft.Json
open OpenAPI.Model.AnyType
open OpenAPI.Model.BasicBotBase
open OpenAPI.Model.BotAllOf
open OpenAPI.Model.int option

module Bot = 

  //#region Bot

  [<CLIMutable>]
  type Bot = {
    [<JsonProperty(PropertyName = "user_id")>]
    UserId : AnyType;
    [<JsonProperty(PropertyName = "full_name")>]
    FullName : AnyType;
    [<JsonProperty(PropertyName = "api_key")>]
    ApiKey : AnyType;
    [<JsonProperty(PropertyName = "default_sending_stream")>]
    DefaultSendingStream : AnyType;
    [<JsonProperty(PropertyName = "default_events_register_stream")>]
    DefaultEventsRegisterStream : AnyType;
    [<JsonProperty(PropertyName = "default_all_public_streams")>]
    DefaultAllPublicStreams : AnyType;
    [<JsonProperty(PropertyName = "avatar_url")>]
    AvatarUrl : AnyType;
    [<JsonProperty(PropertyName = "owner_id")>]
    OwnerId : AnyType;
    [<JsonProperty(PropertyName = "services")>]
    Services : AnyType;
    [<JsonProperty(PropertyName = "email")>]
    Email : string;
    [<JsonProperty(PropertyName = "bot_type")>]
    BotType : int option;
    [<JsonProperty(PropertyName = "is_active")>]
    IsActive : bool;
  }
  
  //#endregion
  